pandas/core/groupby.py

```diff
@@ -2007,8 +2007,9 @@ def group_index(self):
         return self._group_index
 
     def _make_labels(self):
-        if self._was_factor:  # pragma: no cover
-            raise Exception('Should not call this method grouping by level')
+        if self._grouping_type in ("level", "categorical"):  # pragma: no cover
+            raise Exception(
```

Comment:
No, it's not tested anywhere. 

I think it's better to move all making ``self._labels`` and ``self._group_index`` logic from ``__init__`` to this method.